exporter/exporterhelper/logs_batch.go

```diff
@@ -49,11 +49,25 @@ func (req *logsRequest) mergeTo(dst *logsRequest, sz sizer.LogsSizer) {
 
 func (req *logsRequest) split(maxSize int, sz sizer.LogsSizer) []Request {
 	var res []Request
-	for req.size(sz) > maxSize {
-		ld, rmSize := extractLogs(req.ld, maxSize, sz)
-		req.setCachedSize(req.size(sz) - rmSize)
-		res = append(res, newLogsRequest(ld))
+	var ld plog.Logs
+	rmSize := -1
+
+	previousSize := req.size(sz)
+
+	for req.size(sz) > maxSize && rmSize != 0 {
+		ld, rmSize = extractLogs(req.ld, maxSize, sz)
+		if ld.LogRecordCount() > 0 {
+			req.setCachedSize(req.size(sz) - rmSize)
+			res = append(res, newLogsRequest(ld))
+		}
 	}
+
+	if req.size(sz) == previousSize && req.size(sz) > maxSize {
+		// This means we cannot split the log and is not possible
+		// to fit it into the max size.
```

Comment:
Should there be an error, somehow? I'm not following what will happen to the too-large data.

Reply:
In this PR, I drop the data that can not be splitted. If we return an error, the full batch will be dropped. But with this approach, we only drop what can not be splitted.
Maybe is not the best. I’m open to change it :)

Reply:
I like the behavior and the outcome (i.e., drop the data, return success eventually). This is the sort of failure I would place into the `rejected` count in the PartialSuccess message that the OTLP receiver will send on the return path (if synchronous). I realize that not all receivers are OTLP (or synchronous) and that we're talking about _exporter_ behavior here, so it would be simpler to log a warning, like `logger.Warn("individual item larger than size limit", zap.Int("requested", item.Size), zap.Int("limit", config.Limit))`.

I suspect most will agree to a noisy log under the circumstances--otherwise aren't we silently dropping data? Sorry if I've misunderstood, thanks.

Reply:
You understood properly.

Reply:
Done